best-practices/tidb-partitioned-tables-best-practices.md

```diff
@@ -0,0 +1,703 @@
+```
+
+</details>
+
+#### Create a global index on a partitioned table
+
+You can create a global index on a partitioned table using one of the following methods.
+
+> **Note:**
+>
+> - In TiDB v8.5.3 and earlier versions, you can only create global indexes on unique columns. Starting from v8.5.4, TiDB supports global indexes on non-unique columns. This limitation will be removed in a future LTS version.
+> - For non-unique global indexes, use `ADD INDEX` instead of `ADD UNIQUE INDEX`.
+> - You must explicitly specify the `GLOBAL` keyword.
+
+##### Option 1: Use `ALTER TABLE`
+
+To add a global index to an existing partitioned table, use `ALTER TABLE`:
+
+```sql
+ALTER TABLE <table_name>
+ADD UNIQUE INDEX <index_name> (col1, col2) GLOBAL;
+```
+
+##### Option 2: Define the index at table creation
+
+To create a global index when creating a table, define the global index inline in the `CREATE TABLE` statement:
+
+```sql
+CREATE TABLE t (
+  id BIGINT NOT NULL,
+  col1 VARCHAR(50),
+  col2 VARCHAR(50),
+  -- other columns...
+  UNIQUE GLOBAL INDEX idx_col1_col2 (col1, col2)
+)
+PARTITION BY RANGE (id) (
+  PARTITION p0 VALUES LESS THAN (10000),
+  PARTITION p1 VALUES LESS THAN (20000),
+  PARTITION pMax VALUES LESS THAN MAXVALUE
+);
+```
+
+#### Summary
+
+The performance overhead of partitioned tables in TiDB depends significantly on the number of partitions and the type of index used.
+
+- The more partitions you have, the more severe the potential performance degradation.
+- With a smaller number of partitions, the impact might not be noticeable, but it is still workload-dependent.
+- For local indexes, if a query does not include effective partition pruning conditions, the number of partitions directly correlates with the number of [Remote Procedure Calls (RPCs)](https://docs.pingcap.com/tidb/stable/glossary/#remote-procedure-call-rpc) triggered. This means more partitions will likely result in more RPCs, leading to higher latency.
+- For global indexes, the number of RPCs and the degree of performance regression depend on both the number of partitions involved and how many rows need to be retrieved (that is, the number of rows requiring table lookups). Note that for very large tables where data is already distributed across many Regions, accessing data through a global index might have similar performance to a non-partitioned table, as both scenarios require multiple cross-Region RPCs.
```

Comment:
```suggestion
- **Global indexes**: the performance depends on both the number of partitions involved and the number of rows that require table lookups. For very large tables where data is distributed across multiple Regions, accessing data through a global index provides performance similar to that of a non-partitioned table, because both scenarios involve multiple cross-Region RPCs.
```